src/modules/List.js

```diff
@@ -1,53 +1,63 @@
-import ListProperties from './ListProperties.js';
-import verticalDotsIcon from '../img/vertical dots.svg';
-import refresh from '../img/refresh.svg';
+import NewTask from './NewTask.js';
+import ListItem from './List item.js';
 
 export default class List {
-    ListObjects = [
-      new ListProperties(
-        'complete the project',
-        true,
-        0,
-      ),
+  constructor() {
+    this.ListObjects = localStorage.getItem('list') === null ? [] : JSON.parse(localStorage.getItem('list'));
+  }
 
-      new ListProperties(
-        'read the book',
-        true,
-        1,
-      ),
+  addTask() {
+    const addTaskInput = document.querySelector('.add-task');
+    const listform = document.querySelector('.add__task');
+    if (addTaskInput.value.trim().length === 0) return;
+    this.ListObjects.push(new NewTask(addTaskInput.value, false));
+    listform.reset();
+    localStorage.setItem('list', JSON.stringify(this.ListObjects));
+    this.render();
+  }
 
-      new ListProperties(
-        'learn more about react',
-        false,
-        2,
-      ),
-    ]
+  selectTask(event, listLi, verticalDotsIcon, trashIcon) {
+    if (event.target.classList.contains('list-description')) {
+      listLi.classList.toggle('selected');
+      trashIcon.classList.toggle('hidden');
+      this.editTask(event.target);
+    } else if (event.target.classList.contains('trash-icon')) {
+      this.deleteTask(listLi, trashIcon);
+    }
+  }
 
-    render() {
-      const listBody = document.querySelector('.tasks-body');
-      const listTitle = document.querySelector('.task-list');
-      const refreshIcon = new Image();
-      refreshIcon.src = refresh;
-      listTitle.appendChild(refreshIcon);
+  deleteTask(listLi, trashIcon) {
+    listLi.remove();
+    this.ListObjects.splice(trashIcon.id, 1);
+    localStorage.setItem('list', JSON.stringify(this.ListObjects));
+    this.render();
+  }
```

Comment:
- [ ] Good job on removing functionality it's just missing when you remove a task we should re-order indices for example if we have 4 tasks and you remove 3rd one then we should have indices left like 1 2 3

Reply:
@Amrhub yes it fixes the indices after removing tasks from the list that is why I am invoking the render method at the end of the deleteTask method if you use console log at the end of the delete task method you will see that.

![Screenshot_2022-04-01_19-41-07](https://user-images.githubuserco
![deleting tasks](https://user-images.githubusercontent.com/97116942/161314763-893819d2-e61f-4c0c-a0a4-ce8c7baba52b.png)
ntent.com/97116942/161314753-e393c8e4-0b88-469e-ad17-daf736419171.png)

Reply:
Oh I re-checked you are right, I think there was a bug in my browser